base/tools/src/main/java/com/netscape/cmstools/CRMFPopClient.java

```diff
@@ -467,10 +477,33 @@ public static void main(String args[]) throws Exception {
                 subject.addElement(new AVA(new OBJECT_IDENTIFIER("0.9.2342.19200300.100.1.1"),  new PrintableString("MyUid")));
             }
 
-            if (verbose) System.out.println("Generating key pair");
+            if (verbose) System.out.println("Generating key pair: temporary: " + temporary);
             KeyPair keyPair;
             if (algorithm.equals("rsa")) {
-                keyPair = CryptoUtil.generateRSAKeyPair(token, keySize, temporary);
+
+                org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages[] = {
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.ENCRYPT,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.DECRYPT,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.WRAP,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.UNWRAP,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN_RECOVER
+                    };
+
+                    org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages_mask[] = {
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.ENCRYPT,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.DECRYPT,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.WRAP,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.UNWRAP,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN,
+                        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN_RECOVER
+                    };
```

Comment:
Similarly, if these are constants we might want to convert them into class/instance attributes.